### PR TITLE
feat: Spell 2026-08-13 (SC-1605)

### PR DESCRIPTION
**Pull request by supercontracts**
# 2026-08-13 Spark Spell

## Forum Post
https://forum.skyeco.com/t/august-13-2026-proposed-changes-to-spark-for-upcoming-spell/28135#p-106996-summary-1

## Dependencies on Sky Core spell
None

## Notes for Reviewers
None

## Conditions for Spell Deployment
- [ ] All spell-specific tests are passing.
- [ ] All E2E tests are passing.
- [ ] All newly deployed addresses have been reviewed and added to the spark-address-registry in a PR that has been merged.
- [ ] All newly deployed addresses have been added to the Immunefi bug bounty program.
- [ ] There is complete test coverage of all changes in the spell, including any relevant edge cases.
- [ ] At least three approvals have been given from reviewers.
- [ ] CODE_RABBIT.md comment has been posted, AI review performed.
- [ ] Octane review performed.

## Conditions for Spell Handoff to Sky
- [ ] All spells payloads have been deployed and verified on Etherscan.
- [ ] All spell-specific tests are passing (including spell bytecode verification).
- [ ] All E2E tests are passing.
- [ ] Spell caster is passing in CI.
- [ ] At least three approvals have been given from reviewers on deployed spell.
- [ ] Comment from at least three reviewers confirming address of mainnet spell payload.

## Conditions for Spell Merge
- [ ] Spell handoff performed to Sky team on Discord.
- [ ] Spell handoff performed to Sky team on Signal.
- [ ] Spell address confirmed to be correct in sky-ecosystem/spells-mainnet repo open PR through comments by at least three reviewers.
- [ ] Sky deployed spell confirmed through a comment in this PR to contain correct Spark payload address by at least three reviewers.

## Summary by CodeRabbit

* **New Features**
  * Added July 16 Spark updates across Ethereum, Robinhood, and X Layer.
  * Enabled USDG and USDT bridging, updated transfer limits, and configured cross-chain recipients.
  * Updated liquidity-layer permissions and disabled deposits and withdrawals for the legacy USDT vault.
  * Added August 13 liquidity integrations for Uniswap V4 and Curve pools supporting USDG, USDS, and RLUSD.
  * Allocated excess USDS to support SPK buybacks and distributed treasury incentives and operational funds.

### CODE REVIEW

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        bytes32 withdrawPoolId = keccak256(abi.encode(controller.LIMIT_UNISWAP_V4_WITHDRAW(), USDG_USDS_POOL_ID));
+        bytes32 swapPoolId     = keccak256(abi.encode(controller.LIMIT_UNISWAP_V4_SWAP(),     USDG_USDS_POOL_ID));
+
+        assertEq(controller.maxSlippages(address(uint160(uint256(USDG_USDS_POOL_ID)))), 0);
+
+        _assertRateLimit(depositPoolId,  0, 0);
+        _assertRateLimit(withdrawPoolId, 0, 0);
+        _assertRateLimit(swapPoolId,     0, 0);
+
+        (int24 _tickLowerMin, int24 _tickUpperMax, uint24 _maxTickSpacing) = controller.uniswapV4TickLimits(USDG_USDS_POOL_ID);
```
Comment (TheMj0ln1r) [inline src/proposals/20260813/Spell_20260813.t.sol]: space after bracket. (Apply to entire file)

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+        assertEq(_tickLowerMin,   0);
+        assertEq(_tickUpperMax,   0);
+        assertEq(_maxTickSpacing, 0);
+
+        _executeAllPayloadsAndBridges();
+
+        assertEq(controller.maxSlippages(address(uint160(uint256(USDG_USDS_POOL_ID)))), 0.999e18);
+
+        _assertRateLimit(depositPoolId,  10_000_000e18,      100_000_000e18 / uint256(1 days));
+        _assertRateLimit(withdrawPoolId, type(uint256).max,  0);
```
Comment (TheMj0ln1r) [inline src/proposals/20260813/Spell_20260813.t.sol]: extra spaces

**Review by TheMj0ln1r — CHANGES_REQUESTED** [review]
lgtm, just few formats.

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+    uint256 internal constant _V4_SWAP                 = 0x10;
+
+    address internal constant _STATE_VIEW = 0x7fFE42C4a5DEeA5b0feC41C94C136Cf115597227;
+
+    constructor() {
+        _spellId   = 20260813;
+        _blockDate = 1785761559;
+    }
+
+    function test_ETHEREUM_sparkLiquidityLayer_onboardUniswapV4USDGUSDS() public onChain(ChainIdUtils.Ethereum()) {
```
Comment (unvariant-dev) [inline src/proposals/20260813/Spell_20260813.t.sol:37]: Tests don't check the pool ids, would be nice to add something like:

```solidity
function test_ETHEREUM_uniswapV4PoolIdIntegrity() public onChain(ChainIdUtils.Ethereum()) {
    _assertUniswapV4PoolId(USDG_USDS_POOL_ID,  Ethereum.USDS, Ethereum.USDG);
    _assertUniswapV4PoolId(RLUSD_USDS_POOL_ID, RLUSD,         Ethereum.USDS);
}

function _assertUniswapV4PoolId(bytes32 poolId, address currency0, address currency1) internal view {
    PoolKey memory expectedPoolKey = PoolKey({
        currency0   : Currency.wrap(currency0),
        currency1   : Currency.wrap(currency1),
        fee         : 5,  // 0.0005%
        tickSpacing : 1,
        hooks       : IHooks(address(0))
    });

    assertEq(keccak256(abi.encode(expectedPoolKey)), poolId);
}
```

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+
+        assertEq(_tickLowerMin,   -10);
+        assertEq(_tickUpperMax,   10);
+        assertEq(_maxTickSpacing, 10);
+
+        _testUniswapV4LimitOrder(RLUSD_USDS_POOL_ID);
+    }
+
+    function test_ETHEREUM_curvePoolOnboarding_RLUSDUSDC() external onChain(ChainIdUtils.Ethereum()) {
+        _testCurveOnboarding({
```
Comment (unvariant-dev) [inline src/proposals/20260813/Spell_20260813.t.sol:113]: Worth also asserting the coins here:

```solidity
assertEq(ICurvePoolLike(CURVE_RLUSD_USDC).coins(0), Ethereum.USDC);
assertEq(ICurvePoolLike(CURVE_RLUSD_USDC).coins(1), RLUSD);
```

File: src/test-harness/SparkLiquidityLayerTests.sol
Comment (unvariant-dev) [inline src/test-harness/SparkLiquidityLayerTests.sol:1367]: Consider adding a negative test for disabled Curve deposit/withdraw:

```solidity
    uint256[] memory zeroAmounts = new uint256[](2);
    zeroAmounts[0] = 1;
    zeroAmounts[1] = 1;

    vm.prank(vars.ctx.relayer);
    vm.expectRevert("MC/min-amount-not-met");
    vars.controller.addLiquidityCurve(pool, zeroAmounts, 0);

    vm.prank(vars.ctx.relayer);
    vm.expectRevert("MC/min-amount-not-met");
    vars.controller.removeLiquidityCurve(pool, 1, zeroAmounts);
```

File: src/proposals/20260813/SparkEthereum_20260813.sol [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+/**
+ * @title  August 13, 2026 Spark Ethereum Proposal
+ * @author Phoenix Labs
+ * @notice Spark Liquidity Layer:
+ *         - Onboard Uniswap v4 USDG/USDS Pool.
+ *         - Onboard Uniswap v4 rlUSD/USDS Pool.
+ *         - Onboard Curve rlUSD/USDC for Swaps.
+ *         Spark Treasury:
+ *         - Transfer Excess USDS for Buybacks.
+ * Forum:  https://forum.skyeco.com/t/august-13-2026-proposed-changes-to-spark-for-upcoming-spell/28135/8
```
Comment (lucas-manuel) [inline src/proposals/20260813/SparkEthereum_20260813.sol]: Update to use first forum post

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+        _spellId   = 20260813;
+        _blockDate = 1785761559;
+    }
+
+    function test_ETHEREUM_sparkLiquidityLayer_onboardUniswapV4USDGUSDS() public onChain(ChainIdUtils.Ethereum()) {
+        MainnetController controller = MainnetController(Ethereum.ALM_CONTROLLER);
+
+        bytes32 depositPoolId  = keccak256(abi.encode(controller.LIMIT_UNISWAP_V4_DEPOSIT(),  USDG_USDS_POOL_ID));
+        bytes32 withdrawPoolId = keccak256(abi.encode(controller.LIMIT_UNISWAP_V4_WITHDRAW(), USDG_USDS_POOL_ID));
+        bytes32 swapPoolId     = keccak256(abi.encode(controller.LIMIT_UNISWAP_V4_SWAP(),     USDG_USDS_POOL_ID));
```
Comment (lucas-manuel) [inline src/proposals/20260813/Spell_20260813.t.sol]: depositKey, withdrawKey, swapKey, apply to all

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+
+            ( uint256 withdrawnAmount0, ) = _removeLiquidity(poolId, tokenId);
+
+            assertApproxEqRel(
+                _toNormalizedAmount(poolKey.currency1, depositedAmount1),
+                _toNormalizedAmount(poolKey.currency0, withdrawnAmount0),
+                0.001e18
+            );
+        }
+    }
```
Comment (lucas-manuel) [inline src/proposals/20260813/Spell_20260813.t.sol]: Move to a dedicated UniV4Helpers library

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+            ( , uint256 withdrawnAmount1 ) = _removeLiquidity(poolId, tokenId);
+
+            assertApproxEqRel(
+                _toNormalizedAmount(poolKey.currency0, depositedAmount0),
+                _toNormalizedAmount(poolKey.currency1, withdrawnAmount1),
+                0.001e18
+            );
+        }
+
+        if (placeLimitOrderSellToken0 && placeLimitOrderSellToken1) {
```
Comment (lucas-manuel) [inline src/proposals/20260813/Spell_20260813.t.sol]: Add a comment to say that if both conditions are true the rate limits need to be replenished in order to do the second swap

File: src/proposals/20260813/SparkEthereum_20260813.sol [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+ * Vote:   https://snapshot.org/#/s:sparkfi.eth/proposal/0x101ed4ff7ef77f8e4a6db6af4bbf053b703001c0245a1e166babb8ba6ad633fb
+ *         https://snapshot.org/#/s:sparkfi.eth/proposal/0xf99d50e34900ea8f54c90053584f758d2a6d1ddbbe77a8c8d751e2c5a8fd0493
+ *         https://snapshot.org/#/s:sparkfi.eth/proposal/0x3240fc78276a2f4898188809464b3357124b2c42065f55a46a7a7254eabd0f82
+ */
+contract SparkEthereum_20260813 is SparkPayloadEthereum {
+
+    bytes32 internal constant USDG_USDS_POOL_ID  = 0x28adc7179a8a83c3379955d59563c0fec33eadfa83946b447af289190ff5fcff;
```
Comment (lucas-manuel) [inline src/proposals/20260813/SparkEthereum_20260813.sol]: Needs to be USDS_USDG ordering everywhere

File: src/test-harness/SparkLiquidityLayerTests.sol [hunk 10]
```diff
@@ -515,6 +515,11 @@ abstract contract SparkLiquidityLayerTests is SpellRunner {
     bytes32 internal constant PYUSD_USDS_POOL_ID = 0xe63e32b2ae40601662f760d6bf5d771057324fbd97784fe1d3717069f7b75d45;
     bytes32 internal constant USDT_USDS_POOL_ID  = 0x3b1b1f2e775a6db1664f8e7d59ad568605ea2406312c11aef03146c0cf89d5b9;
 
+    bytes32 internal constant USDG_USDS_POOL_ID  = 0x28adc7179a8a83c3379955d59563c0fec33eadfa83946b447af289190ff5fcff;
```
Comment (lucas-manuel) [inline src/test-harness/SparkLiquidityLayerTests.sol]: Needs to be USDS_USDG ordering

File: src/proposals/20260813/SparkEthereum_20260813.sol [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    bytes32 internal constant RLUSD_USDS_POOL_ID = 0x9035721b23481db3888fd201b9c2b26dbc3af60258bca65e669f2ed98dc8eb4f;
+
+    address internal constant CURVE_RLUSD_USDC = 0xD001aE433f254283FeCE51d4ACcE8c53263aa186;
```
Comment (lucas-manuel) [inline src/proposals/20260813/SparkEthereum_20260813.sol]: USDC_RLUSD

File: src/test-harness/SparkLiquidityLayerTests.sol [hunk 12]
```diff
@@ -515,6 +515,11 @@ abstract contract SparkLiquidityLayerTests is SpellRunner {
...
+    bytes32 internal constant RLUSD_USDS_POOL_ID = 0x9035721b23481db3888fd201b9c2b26dbc3af60258bca65e669f2ed98dc8eb4f;
+
+    address internal constant CURVE_RLUSD_USDC = 0xD001aE433f254283FeCE51d4ACcE8c53263aa186;
```
Comment (lucas-manuel) [inline src/test-harness/SparkLiquidityLayerTests.sol]: USDC_RLUSD

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Reviewed spell, LGTM after these changes.

**Review by TheMj0ln1r — APPROVED** [review]

**Review by TheMj0ln1r — DISMISSED** [review]
Approving the Spell for deployment

- [x]  Check forum post link - https://forum.skyeco.com/t/august-13-2026-proposed-changes-to-spark-for-upcoming-spell/28135/8
- [x]  Check description for all payloads
- [x]  Check poll links
- [x]  Check spark-address-registry commit matches current HEAD
- SLL - Uniswap V4 SLL Onboarding [Ethereum] - [1] - DONE
    - [x]  Verify Pool ID in Uniswap UI - `USDG_USDS_POOL_ID=0x28adc7179a8a83c3379955d59563c0fec33eadfa83946b447af289190ff5fcff`
    - [x]  Independently convert ticks to prices from forum post to validate correctness
        
        ```solidity
        >>> (1.0001 ** -276_334) * 10**12
        0.9990031909716848
        >>> (1.0001 ** -276_314) * 10**12
        1.0010030965990384
        ```
        
    - [x]  Review all parameters in `configureUniswapV4Pool` against forum post
    - [x]  Review `configureUniswapV4Pool` function in SLLHelpers, ensuring that
        - [x]  Tick limits are set - `-276_334` , `-276_314`
        - [x]  Max slippage is set - `0.999e18` (0.1%)
        - [x]  Rate limits are set
            - [x]  Deposit
                - [x]  maxAmount - `10_000_000e18`
                - [x]  slope - `100_000_000e18/day`
            - [x]  Withdraw
                - [x]  maxAmount - unlimited
                - [x]  slope - unlimited
            - [x]  Swap
                - [x]  maxAmount - `5_000_000e18`
                - [x]  slope - `200_000_000e18/day`
    - [x]  Review test - `test_ETHEREUM_sparkLiquidityLayer_onboardUniswapV4USDGUSDS`
        - [x]  Check rate limits have been set
        - [x]  Check maxSlippage has been set
        - [x]  Check tick limits have been set
        - [x]  Check that E2E testing is properly conducting deposit/withdraw/swap with proper state assertions and rate limit checks
- SLL - Uniswap V4 SLL Onboarding [Ethereum] - [2] - DONE
    - [x]  Verify Pool ID in Uniswap UI - `RLUSD_USDS_POOL_ID=0x9035721b23481db3888fd201b9c2b26dbc3af60258bca65e669f2ed98dc8eb4f`
    - [x]  Independently convert ticks to prices from forum post to validate correctness
        
        ```solidity
        >>> (1.0001 ** -10)
        0.9990005497800716
        >>> (1.0001 ** 10)
        1.001000450120021
        ```
        
    - [x]  Review all parameters in `configureUniswapV4Pool` against forum post
    - [x]  Review `configureUniswapV4Pool` function in SLLHelpers, ensuring that
        - [x]  Tick limits are set - `-10`, `10`
        - [x]  Max slippage is set - `0.999e18` (0.1%)
        - [x]  Rate limits are set
            - [x]  Deposit
                - [x]  maxAmount - `10_000_000e18`
                - [x]  slope - `50_000_000e18/day`
            - [x]  Withdraw
                - [x]  maxAmount - unlimited
                - [x]  slope - unlimited
            - [x]  Swap
                - [x]  maxAmount - `5_000_000e18`
                - [x]  slope - `100_000_000e18/day`
    - [x]  Review test - `test_ETHEREUM_sparkLiquidityLayer_onboardUniswapV4RLUSDUSDS`
        - [x]  Check rate limits have been set
        - [x]  Check maxSlippage has been set
        - [x]  Check tick limits have been set
        - [x]  Check that E2E testing is properly conducting deposit/withdraw/swap with proper state assertions and rate limit checks
- SLL - Curve SLL onboarding [Ethereum] - [3] - DONE
    - [x]  Check pool address against forum post  - `CURVE_RLUSD_USDC = 0xD001aE433f254283FeCE51d4ACcE8c53263aa186`
    - [x]  Ensure that the pool is of type `stableswap-ng` (check [curve.fi](http://curve.fi/) UI)
    - [x]  Check that pool address is verified
    - [x]  Ensure that the pool has been seeded with LP shares burned at `address(1)`
    - [x]  Ensure that maxSlippage is set according to forum post - `0.999e18`
    - [x]  Check maxAmount against forum post - `5_000_000e18`
    - [x]  Check slope against forum post - `25_000_000e18/day`
    - [x]  Verify the precision of the rate limit is 1e18
    - [x]  Check rate limit before and after state in a test - `test_ETHEREUM_curvePoolOnboarding_RLUSDUSDC`
    - [x]  Ensure that `_configureCurvePool` helper function is used in spell
    - [x]  Ensure that `_testCurveOnboarding` helper test is used in test
- Withdraw reserves [Ethereum] - [4] - DONE
    - [x]  Review SparkPayloadEthereum code for claiming all reserves
    - [x]  Ensure stablecoins go to ALMProxy and everything else goes to Liquidation multisig
    - [x]  Review that test decrements and increments funds from correct addresses for all reserves
- Treasury Transfer [Ethereum] - [5] - DONE
    - [x]  Check receiver address against forum post or address registry - `ALM_OPS_MULTISIG`
    - [x]  Check amount against forum post - `1_756_359e18`
    - [x]  Review that test decrements and increments funds from correct addresses - `test_ETHEREUM_sparkTreasury_transferExcessUSDSForBuybacks`

File: src/proposals/20260813/Spell_20260813.t.sol (hunk 3 above)
Comment (lucas-manuel) [reply src/proposals/20260813/Spell_20260813.t.sol:37]: Agree

File: src/proposals/20260813/Spell_20260813.t.sol (hunk 4 above)
Comment (lucas-manuel) [reply src/proposals/20260813/Spell_20260813.t.sol:113]: Agree

File: src/test-harness/SparkLiquidityLayerTests.sol
Comment (lucas-manuel) [reply src/test-harness/SparkLiquidityLayerTests.sol:1367]: All good here

File: src/proposals/20260813/Spell_20260813.t.sol (hunk 3 above)
Comment (supercontracts) [reply src/proposals/20260813/Spell_20260813.t.sol:37]: Added

File: src/proposals/20260813/Spell_20260813.t.sol (hunk 4 above)
Comment (supercontracts) [reply src/proposals/20260813/Spell_20260813.t.sol:113]: Added

**Review by lucas-manuel — DISMISSED** [review]
Good to deploy.

- [x]  Check forum post link - https://forum.skyeco.com/t/august-13-2026-proposed-changes-to-spark-for-upcoming-spell/28135/8
- [x]  Check description for all payloads
- [x]  Check poll links
- [x]  Check spark-address-registry commit matches current HEAD
- SLL - Uniswap V4 SLL Onboarding [Ethereum] - [1] DONE
    - [x]  Verify Pool ID in Uniswap UI - `USDG_USDS_POOL_ID=0x28adc7179a8a83c3379955d59563c0fec33eadfa83946b447af289190ff5fcff`
    - [x]  Verify pool is seeded through UI
    - [x]  Check pool token ordering through Etherscan and fee
    - [x]  Independently convert ticks to prices from forum post to validate correctness
        
        ```solidity
        >>> (1.0001 ** -276_334) * 10**12
        0.9990031909716848
        >>> (1.0001 ** -276_314) * 10**12
        1.0010030965990384
        ```
        
    - [x]  Review all parameters in `configureUniswapV4Pool` against forum post
    - [x]  Review `configureUniswapV4Pool` function in SLLHelpers, ensuring that
        - [x]  Tick limits are set - `-276_334` , `-276_314`
        - [x]  Max slippage is set - `0.999e18` (0.1%)
        - [x]  Rate limits are set
            - [x]  Deposit
                - [x]  maxAmount - `10_000_000e18`
                - [x]  slope - `100_000_000e18/day`
            - [x]  Withdraw
                - [x]  maxAmount - unlimited
                - [x]  slope - unlimited
            - [x]  Swap
                - [x]  maxAmount - `5_000_000e18`
                - [x]  slope - `200_000_000e18/day`
    - [x]  Review test - `test_ETHEREUM_sparkLiquidityLayer_onboardUniswapV4USDGUSDS`
        - [x]  Check rate limits have been set
        - [x]  Check maxSlippage has been set
        - [x]  Check tick limits have been set
        - [x]  Check that E2E testing is properly conducting deposit/withdraw/swap with proper state assertions and rate limit checks
- SLL - Uniswap V4 SLL Onboarding [Ethereum] - [2] - DONE
    - [x]  Verify Pool ID in Uniswap UI - `RLUSD_USDS_POOL_ID=0x9035721b23481db3888fd201b9c2b26dbc3af60258bca65e669f2ed98dc8eb4f`
    - [x]  Verify pool is seeded through UI
    - [x]  Check pool token ordering through Etherscan and fee
    - [x]  Independently convert ticks to prices from forum post to validate correctness
        
        ```solidity
        >>> (1.0001 ** -10)
        0.9990005497800716
        >>> (1.0001 ** 10)
        1.001000450120021
        ```
        
    - [x]  Review all parameters in `configureUniswapV4Pool` against forum post
    - [x]  Review `configureUniswapV4Pool` function in SLLHelpers, ensuring that
        - [x]  Tick limits are set - `-10`, `10`
        - [x]  Max slippage is set - `0.999e18` (0.1%)
        - [x]  Rate limits are set
            - [x]  Deposit
                - [x]  maxAmount - `10_000_000e18`
                - [x]  slope - `50_000_000e18/day`
            - [x]  Withdraw
                - [x]  maxAmount - unlimited
                - [x]  slope - unlimited
            - [x]  Swap
                - [x]  maxAmount - `5_000_000e18`
                - [x]  slope - `100_000_000e18/day`
    - [x]  Review test - `test_ETHEREUM_sparkLiquidityLayer_onboardUniswapV4RLUSDUSDS`
        - [x]  Check rate limits have been set
        - [x]  Check maxSlippage has been set
        - [x]  Check tick limits have been set
        - [x]  Check that E2E testing is properly conducting deposit/withdraw/swap with proper state assertions and rate limit checks
- SLL - Curve SLL onboarding [Ethereum] - [3] - DONE
    - [x]  Check pool address against forum post  - `CURVE_RLUSD_USDC = 0xD001aE433f254283FeCE51d4ACcE8c53263aa186`
    - [x]  Check seeded
    - [x]  Ensure that the pool is of type `stableswap-ng` (check [curve.fi](http://curve.fi/) UI)
    - [x]  Check that pool address is verified
    - [x]  Ensure that the pool has been seeded with LP shares burned at `address(1)`
    - [x]  Ensure that maxSlippage is set according to forum post - `0.999e18`
    - [x]  Check maxAmount against forum post - `5_000_000e18`
    - [x]  Check slope against forum post - `25_000_000e18/day`
    - [x]  Verify the precision of the rate limit is 1e18
    - [x]  Check rate limit before and after state in a test - `test_ETHEREUM_curvePoolOnboarding_RLUSDUSDC`
    - [x]  Ensure that `_configureCurvePool` helper function is used in spell
    - [x]  Ensure that `_testCurveOnboarding` helper test is used in test
- Withdraw reserves [Ethereum] - [4] - DONE
    - [x]  Review SparkPayloadEthereum code for claiming all reserves
    - [x]  Ensure stablecoins go to ALMProxy and everything else goes to Liquidation multisig
    - [x]  Review that test decrements and increments funds from correct addresses for all reserves
- Treasury Transfer [Ethereum] - [5] - DONE
    - [x]  Check receiver address against forum post or address registry - `ALM_OPS_MULTISIG`
    - [x]  Check amount against forum post - `1_756_359e18`
    - [x]  Review that test decrements and increments funds from correct addresses - `test_ETHEREUM_sparkTreasury_transferExcessUSDSForBuybacks`

File: src/proposals/20260813/Spell_20260813.t.sol [hunk 13]
```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.25;
+
+import { IERC20Metadata } from "openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
```
Comment (unvariant-dev) [inline src/proposals/20260813/Spell_20260813.t.sol]: These imports are unused since the refactor in `c1aacec4544e25bfc3c9877a439d7d15f729f733`
```solidity
import { IERC20Metadata } from "openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
import { IRateLimits }       from "spark-alm-controller/src/interfaces/IRateLimits.sol";
import { UniswapV4Lib }      from "spark-alm-controller/src/libraries/UniswapV4Lib.sol";
```

Comment (supercontracts) [reply src/proposals/20260813/Spell_20260813.t.sol]: removed

**Review by lucas-manuel — APPROVED** [review]
Approving payload deployed at `0xc827237CB91Fa8E78B8dfA4F77838eDf924C04e9`

**Review by TheMj0ln1r — APPROVED** [review]
Approving spell deployment at `0xc827237CB91Fa8E78B8dfA4F77838eDf924C04e9`

**Review by unvariant-dev — APPROVED** [review]
Confirming and approving the spell deployment at `0xc827237CB91Fa8E78B8dfA4F77838eDf924C04e9`

Completed checklist for both the development and deployment stages: https://gist.github.com/unvariant-dev/9ec09866fed86c5b96b7f9d8e6035e20
